### PR TITLE
feat: added `debug` flag in the `SuperTokenConfig` in the `init()` for logging

### PR DESCRIPTION
**Pull request by IamMayankThakur**
## Summary of change

- Added `debug` flag in the `init()`. Debug logs will be printed if this flag is set to true.
- The old way of settings the environment variable `DEBUG=com.supertokens` will continue to work.

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [x] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [ ] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/configUtils.ts` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: lib/ts/supertokens.ts [hunk 1]
```diff
@@ -49,6 +50,9 @@ export default class SuperTokens {
     telemetryEnabled: boolean;
 
     constructor(config: TypeInput) {
+        if (config.debug) {
```
Comment (rishabhpoddar) [inline lib/ts/supertokens.ts:53]: - we also want it to work for with the env var like before (backwards compatibility)
- Doing implicit if is not a good idea. Make it `if (config.debug === true)`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "supertokens-node",
```
Comment (rishabhpoddar) [inline package.json:2]: missing tests..?
- Should not log if debug is undefined or false
- Should log if debug is set to true
- Should log if env var is set

File: lib/ts/supertokens.ts [hunk 3]
```diff
@@ -49,6 +50,9 @@ export default class SuperTokens {
...
+            debug.enable("com.supertokens");
```
Comment (rishabhpoddar) [inline lib/ts/supertokens.ts:54]: - bad idea to hard code this when we already have `SUPERTOKENS_DEBUG_NAMESPACE` in the logger.ts file.
- Ideally this boolean is passed to the logger.ts file and that then enables the debug logging.. we do not want to directly interact with the `debug` lib in this file.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: test/config.test.js [hunk 4]
```diff
@@ -1192,6 +1194,64 @@ describe(`configTest: ${printPath("[test/config.test.js]")}`, function () {
         }
     });
 
+    it("testing that the debug mode is set", async function () {
```
Comment (rishabhpoddar) [inline test/config.test.js:1197]: this is not how you test if the logging is working.. you need to actually listen to the console and see if it outputs in there?

File: lib/ts/supertokens.ts [hunk 5]
```diff
@@ -48,7 +48,16 @@ export default class SuperTokens {
 
     telemetryEnabled: boolean;
 
+    debugEnabled: boolean;
```
Comment (rishabhpoddar) [inline lib/ts/supertokens.ts:51]: this variable should not be required, and even if it is, it should be in the logger.ts file

File: lib/ts/logger.ts [hunk 6]
```diff
@@ -23,12 +23,14 @@ const SUPERTOKENS_DEBUG_NAMESPACE = "com.supertokens";
 */
 
 function logDebugMessage(message: string) {
-    if (debug.enabled(SUPERTOKENS_DEBUG_NAMESPACE)) {
-        debug(SUPERTOKENS_DEBUG_NAMESPACE)(
-            `{t: "${new Date().toISOString()}", message: \"${message}\", file: \"${getFileLocation()}\" sdkVer: "${version}"}`
-        );
-        console.log();
-    }
+    debug(SUPERTOKENS_DEBUG_NAMESPACE)(
```
Comment (rishabhpoddar) [inline lib/ts/logger.ts:26]: why aren't you checking for `if (debug.enabled(SUPERTOKENS_DEBUG_NAMESPACE)) {`?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/logger.ts (hunk 6 above)
Comment (IamMayankThakur) [reply lib/ts/logger.ts:26]: We already call `debug.enable()` when initialising the logger, so ideally this should always return `true`. Added this check, just for safety.

File: lib/ts/supertokens.ts (hunk 5 above)
Comment (IamMayankThakur) [reply lib/ts/supertokens.ts:51]: It was only used in the tests, removed it.

File: test/config.test.js (hunk 4 above)
Comment (IamMayankThakur) [reply test/config.test.js:1197]: Done. overrode the `debug.log` to capture logs.

**Pull request merged**